### DISCUSSION TITLE
Federation sample issue

### DISCUSSION THREAD

**Discussion by knyahoo** [eclipse-rdf4j/rdf4j#4536] [Q&A]
When trying out the Federation sample code, I got Exception in thread "main" java.lang.NoClassDefFoundError 
at org.eclipse.rdf4j.federated.FedXFactory.withSparqlEndpoint(FedXFactory.java:173)

Is there any configuration I need?

**Comment by abrokenjester** [comment]
I think you may be missing a dependency. The federation module is currently not included in our rdf4j-storage umbrella-PM (a bit of an oversight), so you'll have to explicitly add the following maven dependency to use it:

```xml
<dependency>
    <groupId>org.eclipse.rdf4j</groupId>
    <artifactId>rdf4j-tools-federation</artifactId>
</dependency>
```
